.github/workflows/ci-core.yml

```diff
@@ -53,9 +54,28 @@ jobs:
         id: ignore-filter
         run: echo "changes=true" >> $GITHUB_OUTPUT
 
+      - name: Filter other changes
+        uses: ./.github/actions/filter-changes
+        id: filter
+        with:
+          # Filter files that match **every** pattern in this list
+          every-filter: |
+            - '**'                     # Initially include all files
+            - '!**/.github/**'         # Exclude all changes under the .github directory
+            - '!integration-tests/**'  # Exclude all changes under the integration-tests directory
+          # Filter files that match **any** of these patterns
+          any-filter: |
+            - '.github/workflows/ci-core.yml'
+            - '.github/actions/golangci-lint/**'
+            - '.github/actions/filter-changes/**'
+            - '.github/actions/setup-go/**'
+            - '.github/actions/setup-nodejs/**'
+            - '.github/actions/setup-postgres/**'
+            - '.github/actions/setup-solana/**'
+            - '.github/actions/setup-wasmd/**'
```

Comment:
These are dangerous optimizations, if any of these get renamed / moved, then it'll potentially silently fail. The cost of CI breaking even once is significantly higher than saving what amounts to a paltry amount of CI runs.

Running 
```sh
git rev-list --count --since="4 months ago" HEAD -- .github/actions/
```
on `develop` gives 31. We are risking a silent failure on the most critical part of CI to avoid running on an extra 7.8 commits a month. For reference, within that same 4 month timespan, we have made 864 commits, or 216 commits a month. This is not even a 4% improvement in terms of commits that we'll run this CI on.

Even this is a conservative estimate. When you consider that commits in develop are squash merges, and feature PRs usually do not contain any changes to CI, and account for a significantly larger proportion of pre-squash commits.

Reply:
> These are dangerous optimizations, if any of these get renamed / moved, then it'll potentially silently fail. 

@HenryNguyen5 but didn't you do the same in https://github.com/smartcontractkit/chainlink/pull/15081/files#diff-31bd198ce2c6956ed10bdfe25a233b755f3a201ad094a67c449115ef76d9ff43R69-R87? How's my PR different?